### PR TITLE
[MB-4628] Add Payment Details to Payment Request Card

### PR DESCRIPTION
**Pull request by noelledusahel**
## Description

Given that I am a TIO looking at the payment requests page
When I look at a payment request card for PENDING or REVIEWED requests
Then I see the following information:
- date approved/rejected,
- requesting contractor, contract number, and LOA information.

## Reviewer Notes
Screen Shot 2021-01-07 at 11 17 54 AM

Screen Shot 2021-01-07 at 11 18 02 AM

Screen Shot 2021-01-07 at 6 00 58 PM

Is there anything you would like reviewers to give additional scrutiny?

## Setup

Add any steps or code to run in this section to help others prepare to run your code:

```sh
make db_dev_e2e_populate
make server_run
make office_client_run
```

## Code Review Verification Steps

* [ ] If the change is risky, it has been tested in experimental before merging.
* [ ] Code follows the guidelines for [Logging](https://github.com/transcom/mymove/wiki/Backend-Programming-Guide#logging)
* [ ] The requirements listed in
 [Querying the Database Safely](https://github.com/transcom/mymove/wiki/Backend-Programming-Guide#querying-the-database-safely)
 have been satisfied.
* Any new migrations/schema changes:
  * [ ] Follow our guidelines for zero-downtime deploys (see [Zero-Downtime Deploys](https://github.com/transcom/mymove/wiki/migrate-the-database#zero-downtime-migrations))
  * [ ] Have been communicated to #g-database
  * [ ] Secure migrations have been tested following the instructions in our [docs](https://github.com/transcom/mymove/wiki/migrate-the-database#secure-migrations)
* [ ] There are no aXe warnings for UI.
* [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/master/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
* [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
* [ ] User facing changes have been reviewed by design.
* [ ] Request review from a member of a different team.
* [ ] Have the Jira acceptance criteria been met for this change?

## References

* [Jira story](https://dp3.atlassian.net/secure/RapidBoard.jspa?rapidView=27&modal=detail&selectedIssue=MB-4628) for this change
* [this article](tbd) explains more about the approach used.

## Screenshots

If this PR makes visible UI changes, an image of the finished UI can help reviewers and casual
observers understand the context of the changes. A before image is optional and
can be included at the submitter's discretion.

Consider using an animated image to show an entire workflow instead of using multiple images. You may want to use GIPHY CAPTURE for this! 📸

_Please frame screenshots to show enough useful context but also highlight the affected regions._

### CODE REVIEW

File: src/shared/formatters.js [hunk 1]
```diff
@@ -193,6 +193,19 @@ export function formatDate(date, inputFormat, outputFormat = 'DD-MMM-YY', locale
   }
 }
 
+// Format a date and ignore any time values, e.g. 03 Jan 2018
+export function formatDateWithoutDash(
```
Comment (duncan-truss) [inline src/shared/formatters.js:197]: I think this function can become:
```
export function formatDateWithoutDash(date) {
    return formatDateFromIso(date, 'DD MMM YYYY')
}
```

File: pkg/handlers/ghcapi/internal/payloads/model_to_payload.go [hunk 2]
```diff
@@ -14,18 +15,38 @@ import (
...
+	return payload
+}
+
 // Move payload
 func Move(move *models.Move) *ghcmessages.Move {
 	if move == nil {
 		return nil
 	}
+	fmt.Printf("contractor: %v", move.Contractor)
+	fmt.Printf("contractorID: %v", move.ContractorID)
```
Comment (duncan-truss) [inline pkg/handlers/ghcapi/internal/payloads/model_to_payload.go:40]: Stray debugging lines

**Comment by duncan-truss** [comment]
@noelledusahel In the devseed testdata we have a reviewed Payment Request but one of the payment request service items is not approved or denied.  Can you update the status so this doesn't show as empty?  https://github.com/transcom/mymove/blob/227032fed0b0dab7d13bac82cf1a916271327784/pkg/testdatagen/scenario/devseed.go#L1630

Screen Shot 2021-01-07 at 3 57 48 PM

**Comment by duncan-truss** [comment]
> @noelledusahel In the devseed testdata we have a reviewed Payment Request but one of the payment request service items is not approved or denied. Can you update the status so this doesn't show as empty?
> 
> https://github.com/transcom/mymove/blob/227032fed0b0dab7d13bac82cf1a916271327784/pkg/testdatagen/scenario/devseed.go#L1630
> 
> Screen Shot 2021-01-07 at 3 57 48 PM
> 
> @duncan-truss are we talking update the Payment Request to requested or updating the status on a Pr Service item?

The status of the Payment Request is already Reviewed so that means the service item should be in either the approved or denied status not requested.  So you'd be updating the pr service item status and adding a reviewedAt timestamp to the payment requests which I think was also missing please.

**Comment by noelledusahel** [comment]
> > @noelledusahel In the devseed testdata we have a reviewed Payment Request but one of the payment request service items is not approved or denied. Can you update the status so this doesn't show as empty?
> > https://github.com/transcom/mymove/blob/227032fed0b0dab7d13bac82cf1a916271327784/pkg/testdatagen/scenario/devseed.go#L1630
> > 
> > Screen Shot 2021-01-07 at 3 57 48 PM
> > @duncan-truss are we talking update the Payment Request to requested or updating the status on a Pr Service item?
> 
> @noelledusahel The status of the Payment Request is already Reviewed so that means the service item should be in either the approved or denied status not requested. So you'd be updating the pr service item status and adding a reviewedAt timestamp to the payment requests which I think was also missing please.

ok thanks for clarifying!

File: src/types/moveOrder.js [hunk 3]
```diff
@@ -149,4 +149,6 @@ export const PaymentRequestShape = PropTypes.shape({
   eTag: PropTypes.string,
   serviceItems: PropTypes.arrayOf(PaymentServiceItemShape),
   reviewedAt: PropTypes.string,
+  Sac: PropTypes.string,
```
Comment (noelledusahel) [inline src/types/moveOrder.js:152]: removing this

File: src/components/Office/PaymentRequestCard/PaymentRequestCard.jsx [hunk 4]
```diff
@@ -34,6 +33,15 @@ const PaymentRequestCard = ({ paymentRequest, history }) => {
   let approvedAmount = 0;
   let rejectedAmount = 0;
 
+  let tac;
+  let sac;
+  let contractNumber;
+
+  if (paymentRequest.moveTaskOrder) {
```
Comment (duncan-truss) [inline src/components/Office/PaymentRequestCard/PaymentRequestCard.jsx:40]: I don't think we can have a payment request that doesn't have a move task order.  Did you see bad test data somewhere?

Comment (noelledusahel) [reply src/components/Office/PaymentRequestCard/PaymentRequestCard.jsx:40]: It was specifically a Happo diff error, `can't read orders on undefined`

Comment (duncan-truss) [reply src/components/Office/PaymentRequestCard/PaymentRequestCard.jsx:40]: Ahh ok so maybe update the payment request prop shape in PaymentRequestCard.stories.jsx to add the orders and contractor?

Comment (noelledusahel) [reply src/components/Office/PaymentRequestCard/PaymentRequestCard.jsx:40]: ya that is better

**Review by duncan-truss — APPROVED** [review]
LGTM! It would be good to update the tests with what you've added.

**Review by Ryan-Koch — APPROVED** [review]

**Review by amanda5iu — COMMENTED** [review]
One small copy request: `Contract number` should be sentence case. We should default to sentence case in general, unless there's a clear exception.

There are some visual issues too but I believe they'll be addressed by the Storybook component [from MB-4629](https://github.com/transcom/mymove/pull/5608).

**Review by amanda5iu — APPROVED** [review]

File: src/components/Office/PaymentRequestCard/PaymentRequestCard.test.jsx [hunk 5]
```diff
@@ -112,12 +150,17 @@ describe('PaymentRequestCard', () => {
   });
 
   describe('reviewed payment request', () => {
+    const reviewedAtDate = moment(MockDate);
```
Comment (duncan-truss) [inline src/components/Office/PaymentRequestCard/PaymentRequestCard.test.jsx:153]: @noelledusahel I'm not sure you needed this, was there a reason the reviewedAt field wasn't enough?

**Pull request merged**